lib/explorer/series.ex

```diff
@@ -1161,6 +1173,17 @@ defmodule Explorer.Series do
     end
   end
 
+  def select(%Series{} = predicate, on_true, on_false) do
+    select(predicate, to_series!(on_true), to_series!(on_false))
+  end
+
+  defp to_series!(%Series{} = series), do: series
+
+  defp to_series!(value) do
+    dtype = Explorer.Shared.check_types!([value])
+    from_list([value], dtype: dtype)
+  end
```

Comment:
I am looking at other code in this module and it seems that what we typically do is this:

```suggestion
  def select(%Series{} = predicate, on_true, on_false) do
    apply_series_list(:select, [predicate, on_true, on_false])
  end
```

And the conversion is done by the backend (and not upfront). This is what we do on all binary operators. Perhaps it is worth doing the same?